fem/pfespace.cpp

```diff
@@ -3507,6 +3507,125 @@ void ParFiniteElementSpace::Update(bool want_transform)
    }
 }
 
+void ParFiniteElementSpace::GetTrueUpdateOperator(OperatorHandle &T)
+{
+   bool need_P_old = false;
+   std::unique_ptr<HypreParMatrix> P_old;
+
+   T.Clear();  // preserves T.Type()
+
+   switch (mesh->GetLastOperation())
+   {
+      case Mesh::REFINE:
+      {
+         if (T.Type() == Operator::Hypre_ParCSR)
+         {
+            SetUpdateOperatorType(Operator::MFEM_SPARSEMAT);
+            need_P_old = true;
+         }
+         break;
+      }
+
+      case Mesh::DEREFINE:
+      {
+         if (T.Type() == Operator::Hypre_ParCSR)
+         {
+            need_P_old = true;
+         }
+         break;
+      }
+
+      case Mesh::REBALANCE:
+      {
+         if (T.Type() == Operator::Hypre_ParCSR)
+         {
+            need_P_old = true;
+         }
+         break;
+      }
+
+      default:
```

Comment:
Should there be an `MFEM_ABORT` in the default case, or is there a case that should reach here?

Reply:
Currently the documentation states that an empty `T` will be returned in case of an error.

We can change this current behavior and generate an MFEM error (e.g. with `MFEM_ABORT`) -- I'm not sure if this behavior is preferred from users' perspective. @tepperly, what do you think?

Reply:
Actually, in the current implementation we do generate an MFEM error (with `MFEM_VERIFY`) when we encounter an internal error (an error that should never happen). The cases of user errors (e.g. calling the method with a NURBS mesh or with unsupported `T.Type()`) is when we return empty `T`, i.e. `T` holding `NULL`.

Reply:
Is there a reason to allow calling with an unsupported type, with `NULL` returned? Generating an error means that the user has to be sure to pass a supported type, and I'm not sure why that would be a problem for the caller.

Reply:
Considering `NONE` is the only remaining option in `Operation { NONE, REFINE, DEREFINE, REBALANCE }`, we could just return at the beginning in the `NONE` case?